### DISCUSSION TITLE
Rewrite to Rust lang

### DISCUSSION THREAD

**Discussion by tujoworker** [babel/babel#13993] [Ideas]
It's more a question than an idea. But is there any plans to use a more memory efficient language to handle the core of Babel?

I could not find any? So therefore I ask here 🙏🏻

**Comment by nicolo-ribaudo** [comment]
No, there isn't. If you are interested, there are Rust-based alternatives to Babel such as https://swc.rs/.

**Reply by tujoworker** [reply]
Thanks for the quick reply! I know about swc – but because Babel is awesome, and we not always should re-new our stack and tools every five years – it would be a really nice thing if major players like Babel and Webpack listen and re-new their internals and adapt approaches so they can complete for a long time ahead.

I very much like the approach that Babel can run exclusively in the Browser – which swc can't – but I guess, also this could be solveable – like, run some core functions in Rust, when rust is available. Also, there is Webassembly and approaches to run expensive code in Rust.

Now, I know, its not so easy, especially the part of knowledge around these other programming languages. But one thing is to power this from current core maintainers, and another thing is to plan and create a roadmap so others can be involved to make this happen.

I guess all of this is considered already – and maybe there are some resources I can read more about – why no plans have been made?

@nicolo-ribaudo Thank you for being a maintainer and for all your work 🙏
